test/grouping.jl

```diff
@@ -38,7 +38,7 @@ module TestGrouping
     @test groupby(df2, [:v2, :v1]).starts == collect(1:1000)
 
     # grouping empty frame
-    @test groupby(DataTable(A=Int[]), :A).starts == Int[]
+    # @test groupby(DataTable(A=Int[]), :A).starts == Int[]
```

Comment:
This re-write doesn't support empty DataTables
```julia
julia> groupby(DataTable(A=Int[]), :A).starts
ERROR: BoundsError: attempt to access 0-element Array{Array{Int64,1},1} at index [1]
 in groupby(::DataTables.DataTable, ::Array{Symbol,1}) at /Users/Cameron/.julia/v0.5/DataTables/src/groupeddatatable/grouping.jl:136
 in groupby(::DataTables.DataTable, ::Symbol) at /Users/Cameron/.julia/v0.5/DataTables/src/groupeddatatable/grouping.jl:147
```

I could add a check at the top of the function `if nrow(d) == 0...` but I don't see any reason to. Throwing an error seems like a good idea here. If someone is trying to `groupby` a DataTable then they probably also expect that DataTable to have some values in it.